### DISCUSSION TITLE
[RFC FS-1102] Discussion thread for discards on use bindings

### DISCUSSION THREAD

**Discussion by Happypig375** [fsharp/fslang-design#578] [Language and core library RFC discussions]
RFC: https://github.com/fsharp/fslang-design/blob/master/RFCs/FS-1102-discards-on-use-bindings.md
Suggestion: fsharp/fslang-suggestions#881

**Comment by charlesroddie** [comment]
This RFC doesn't match fsharp/fslang-suggestions#881 (underscore in use bindings) but instead is an extension or follow-on suggestion (pattern matches in use bindings). Usually there is some discussion in fslang-suggestions before posting RFCs. fsharp/fslang-suggestions#881 itself doesn't need an RFC as it's very straightforward.

The behaviour in a follow-on suggestion would need to be discussed/specced. This PR says it should match CEs, but also suggests that tuples of IDisposables should be deconstructed, which don't seem to be possible from CEs.

**Reply by dsyme** [reply]
Yes, this is correct, as spec'd this wasn't approved.  The only thing approved is the use of `use _ = ....`

**Reply by dsyme** [reply]
My comment here: https://github.com/fsharp/fslang-suggestions/issues/881#issuecomment-853807713
